plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -61,6 +61,8 @@ export interface CatalogTableProps {
   tableOptions?: TableProps<CatalogTableRow>['options'];
   emptyContent?: ReactNode;
   subtitle?: string;
+  calculateCounterFunction?: Function,
+  buildSubtitleFunction?: Function
```

Comment:
I would recommend using a more specific type so we know what these functions accept and return:
```suggestion
  calculateCounterFunction?: (entities: Entity[]) => number,
  buildSubtitleFunction?: (entities: Entity[]) => string
```

Reply:
I will iterate with something like:

```
  calculateCounterFunction?: (options: { context: EntityListContextProps }) => number,
  buildSubtitleFunction?: (options: { context: EntityListContextProps }) => string
```

Keep in mind that if pagination is enabled, `entities` will contain only a subset of all entities present in the catalog. Therefore, you might need to consider additional factors when building your logic.